ruby-challenges.rb

```diff
@@ -3,30 +3,110 @@
 
 # --------------------1) Create a method that takes in an array of words and a single letter and returns an array of all the words containing that particular letter. Use the test variables provided.
 
+# Pseudocode:
+# Define a method drink_finder
+# applied to: an array of strings
+# input: a string consisting of a single letter
+# behavior: outputs an array containing only those strings which contain the given letter
+# use the .select method to return a selection of the array
+# use the .include? method to check if the value contains the given letter
+
 beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
+letter_o = 'o'  # Expected output: ['coffee', 'soda water']
+letter_t = 't'  # Expected output: ['tea', 'water', 'soda water']
 
-letter_o = 'o'
-# Expected output: ['coffee', 'soda water']
-letter_t = 't'
-# Expected output: ['tea', 'water', 'soda water']
+def drink_finder(letter, array)
+    array.select {|v| v.include? letter}
+end
+p drink_finder(letter_o, beverages_array)
+p drink_finder(letter_t, beverages_array)
+# ┌[aaron@ACMBP] [/dev/ttys002] [ruby-challenges ⚡] 
+# └[~/Desktop/week-5-assessment-s8acrine]> ruby ruby-challenges.rb
+# ["coffee", "soda water"]
+# ["tea", "water", "soda water"]
 
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
+# Pseudocode:
+# Define a method  state_sorter_a_z
+# Input :(hash containing a set of arrays)
+# Output: One array containing the arrays from the hash, in alphabetical order, with no nested arrays
+# Create an empty array to return
+# Use .each to iterate over each key value pair
+# Use .each AGAIN to iterate over the array inside the hash
+# Use the shovel operator to add values to a new array
+# Use the .sort! method to sort the array in-place for performance
+# Return the array 
+# No nested arrays to even deal with!
+
+
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+def state_sorter_a_z hash
+    states = []
+    hash.each_value {|i| i.each {|j| states << j }  }
+    states.sort!
+end
+
```

Comment:
I like this approach but definitely encourage you to take care of the ruby methods. There's some really dev friendly options 👍🏽